docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -352,12 +371,45 @@ class CustomerBusinessFactory extends SprykerCustomerBusinessFactory
 }
 ```
 
+2. Use the project-level factory method instead of core-level factory method.
+
+```php
+use Spryker\Zed\CustomerAccess\Persistence\CustomerAccessEntityManager as SprykerCustomerAccessEntityManager;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerAccessEntityManager extends SprykerCustomerAccessEntityManager implements CustomCustomerAccessEntityManager
+{
+    /**
+     *
+     */
+    public function setContentTypesToAccessible(CustomerAccessTransfer $customerAccessTransfer): void
+    {
+        $this->getFactory()->createPyzCustomerAccessMapper();
+       ....
+    }
+}
+```
+
+After applying the solution, re-evaluate the code. The same error shouldn’t be returned. As soon as the extension point in core is released, refactor the code using the recommended [development strategies](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html).
+While it's not refactored, auto-upgrades are not supported, and the effort to update the project may be bigger and require more manual work.
+
+---
 
 ## PrivateApi:Dependency
 
 Business factory should use dependency by a key that is defined at the project level.
```

Comment:
Is `Business factory` correct? Sounds strange

Reply:
It's a factory that is located inside of the `Business\` namespace. I check if it is an official naming inside of the Spryker documentation.

Reply:
yes, "Business factory" is already used on other places inside of documentation, so I think it is legal to use it here, example: https://docs.spryker.com/docs/scos/dev/tutorials-and-howtos/advanced-tutorials/tutorial-replacing-a-default-data-importer-with-the-queue-data-importer.html#adjust-the-business-factory